src/averageAges.js

```diff
@@ -55,10 +70,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredWomen = people.filter(person =>
+    person.sex === 'f' && people.some(child => person.name === child.mother)
+  );
+
+  let countOfOperations = 0;
+
+  const sum = filteredWomen.reduce(
+    (total, mother) => {
+      // looking for all the children of the mother
+      const children = people.filter(child => onlyWithSon
+        ? child.mother === mother.name && child.sex === 'm'
+        : child.mother === mother.name
+      );
+
+      // calculate sum of age difference between mother and each of her child
+      return total + children.reduce((totalOfMother, childOfMother) => {
+        countOfOperations++;
+
+        return totalOfMother
+          + childOfMother.born
+          - (people.find(person => person.name === childOfMother.mother).born);
```

Comment:
it seems like you are finding the mother of the child that you were found for this mother a few rows above. Long story short, the `mother` variable (argument of the first reducer) stored the value you trying to get here, so:
```suggestion
          - mother.born;
```

Reply:
Indeed, this is an unnecessary search operation